controllers/link.go

```diff
@@ -0,0 +1,63 @@
+package controllers
+
+import (
+	"net/http"
+
+	"github.com/gin-gonic/gin"
+	"github.com/nimble-link/backend/models"
+	"github.com/nimble-link/backend/pkg/linkutils"
+	"github.com/nimble-link/backend/services/authentication"
+)
+
+type ShortLinkInput struct {
+	OriginalURL string `json:"original_url" binding:"required"`
+	CustomAlias string `json:"custom_alias"`
+	Password    string `json:"password"`
+}
+
+func CreateLink(c *gin.Context) {
+
+	var input ShortLinkInput
+	if err := c.ShouldBindJSON(&input); err != nil {
+		c.AbortWithStatus(http.StatusBadRequest)
+		return
+	}
+
+	// get user info
+	user, err := authentication.GetCurrentUserFromContext(c)
```

Comment:
I think we can use `c.Keys["current_user"]` here for accessing the current user

Reply:
the value of the key "current_user" is user's email right?

Reply:
c.Keys["current_user"] is current user's email right?

Reply:
I still need `user.id` so after get the email, will need to function `FindUserByEmail`, is that okay?

Reply:
https://github.com/nimble-link/backend/blob/5d6f3207fdf2f526b6ff1e5b9763081c92bbebaa/services/authentication/user.go#L24
@suho `c.Keys["current_user"] is an instance of User, not only `email` I think. Btw, you're right, we should use `authentication.GetCurrentUserFromContext(c)` instead of direct accessing  gin's context.